### DISCUSSION TITLE
Removed extra props on `li`

### DISCUSSION THREAD

**Discussion by efattal** [remarkjs/remark#1282] [Q&A]
Hellos guys, 

Since this change https://github.com/remarkjs/react-markdown/blob/main/changelog.md#remove-extra-props-passed-to-certain-components, I wonder how to replace the remove `ordered`, and `index` props that used to be available on li components. Any example of the plugin I should write? How can I gest the parent? Any help is welcome. Thanks a lot.

**Comment by ChristianMurphy** [comment]
Welcome @efattal! 👋 

> I wonder how to replace the remove ordered, and index props that used to be available on li components.

What are you trying to do with the information?
They were removed because there are better alternatives for most situations.
If you are for some reason locked into those specific properties (again I'd question why) you can generate the properties with a rehype plugin, more information on plugins here: https://unifiedjs.com/learn/

> Any example of the plugin I should write?

There are hundreds of examples of working plugins https://github.com/remarkjs/remark/blob/main/doc/plugins.md and https://github.com/rehypejs/rehype/blob/main/doc/plugins.md

> How can I get the parent?

https://github.com/syntax-tree/unist-util-visit-parents

**Reply by efattal** [reply]
Thanks @ChristianMurphy ,

I'm trying to use remarkjs to convert the markdown to plain text. So the extraProps helped me to render the li with bullets or with a number. I'll look at the links you gave me. Thanks a lot.

**Reply by ChristianMurphy** [reply]
Markdown is already plain text 🙂 
Are you looking for something like: https://github.com/remarkjs/remark/tree/main/packages/remark-stringify#readme or https://github.com/syntax-tree/mdast-util-to-string#readme?
If not please take some time to more fully articulate what you are trying to achieve. (https://github.com/remarkjs/.github/blob/main/support.md)
